From 2fa8ffea68498e02005e85c27e61bde30718ae3b Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@whissi.de>
Date: Sat, 11 Jun 2016 16:18:17 +0200
Subject: [PATCH 1/2] configure: Fix detection whether libcurl is linked
 against gnutls

The find_curl macro is also checking whether libcurl is linked against
gnutls. However the check depends on "CURL_LIBS" which wasn't defined
by the macro.

This commit will define "CURL_LIBS" so that the check works as expected.
---
 build/find_curl.m4 | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build/find_curl.m4 b/build/find_curl.m4
index 6b23ad6..3310e40 100644
--- a/build/find_curl.m4
+++ b/build/find_curl.m4
@@ -2,6 +2,7 @@ dnl Check for CURL Libraries
 dnl CHECK_CURL(ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND])
 dnl Sets:
 dnl  CURL_CFLAGS
+dnl  CURL_LDADD
 dnl  CURL_LIBS
 
 CURL_CONFIG=""
@@ -57,7 +58,8 @@ if test -n "${curl_path}"; then
     if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(curl VERSION: $CURL_VERSION); fi
     CURL_CFLAGS="`${CURL_CONFIG} --cflags`"
     if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(curl CFLAGS: $CURL_CFLAGS); fi
-    CURL_LDADD="`${CURL_CONFIG} --libs`"
+    CURL_LIBS="`${CURL_CONFIG} --libs`"
+    CURL_LDADD="${CURL_LIBS}"
     if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(curl LDADD: $CURL_LIBS); fi
 
     dnl # Check version is ok

From 67f98e7da04251a40a0172e3dfac2c5a6ac6f7dd Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@whissi.de>
Date: Sat, 11 Jun 2016 16:48:58 +0200
Subject: [PATCH 2/2] configure: Move verbose_output declaration up to the
 beginning

Macros like "find_curl" are using "verbose_output" variable but because some
of them are called before we define the variable we are seeing errors like

  ./configure: line 13855: test: : integer expression expected

This commit will fix the problem by moving the "verbose_output" declaration
up to the beginning so that the variable is available for every macro.
---
 configure.ac | 30 ++++++++++++++++--------------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7517885..0f32b01 100644
--- a/configure.ac
+++ b/configure.ac
@@ -70,6 +70,22 @@ AC_SUBST(MSC_REGRESSION_DOCROOT_DIR)
 
 ### Configure Options
 
+# Verbose output
+AC_ARG_ENABLE(verbose-output,
+              AS_HELP_STRING([--enable-verbose-output],
+                             [Enable more verbose configure output.]),
+[
+  if test "$enableval" != "no"; then
+    verbose_output=1
+  else
+    verbose_output=0
+  fi
+],
+[
+  verbose_output=0
+])
+
+
 #OS type
 
 AC_CANONICAL_HOST
@@ -410,20 +426,6 @@ AC_ARG_ENABLE(errors,
   report_errors=1
 ])
 
-# Verbose output
-AC_ARG_ENABLE(verbose-output,
-              AS_HELP_STRING([--enable-verbose-output],
-                             [Enable more verbose configure output.]),
-[
-  if test "$enableval" != "no"; then
-    verbose_output=1
-  else
-    verbose_output=0
-  fi
-],
-[
-  verbose_output=0
-])
 
 # Strict Compile
 AC_ARG_ENABLE(strict-compile,
